pkg/interface/publish/src/js/components/lib/dropdown.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+
+export class Dropdown extends Component {
+  constructor(props) {
+    super(props);
+
+    this.toggleDropdown = this.toggleDropdown.bind(this);
+    this.handleClickOutside = this.handleClickOutside.bind(this);
+    this.collapseAndDispatch = this.collapseAndDispatch.bind(this);
+    this.state = {
+      open: false
+    }
+  }
+
+  componentDidMount() {
+    document.addEventListener('mousedown', this.handleClickOutside);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside);
+  }
+
+  handleClickOutside(evt) {
+    if (this.optsList && !this.optsList.contains(evt.target) &&
+        this.optsButton && !this.optsButton.contains(evt.target)) {
+      this.setState({open: false});
+    }
+  }
+
+  toggleDropdown() {
+    this.setState({open: !this.state.open});
+  }
+
+  collapseAndDispatch(action){
+    this.setState({open: false}, action);
+  }
+
+  render() {
+    let display = (this.state.open)
+      ? "block" : "none";
+
+    let optionsColor = (this.state.open)
+      ? '#e6e6e6' : 'white';
+
+    let optionsList = this.props.options.map((val, i) => {
+      return (
+        <button key={i} className={val.cls}
+          onClick={() => this.collapseAndDispatch(val.action)}>
+          {val.txt}
+        </button>
+      );
+    });
+
+    return (
+      <div className="options relative dib"
+           ref={(el) => {this.optsButton = el}}>
+        <button className="pr3 mb1 pointer br2 pa2 pr4"
+                style={{backgroundColor: optionsColor}}
+                onClick={this.toggleDropdown}>
+          {this.props.buttonText}
+        </button>
+        <div className="absolute flex flex-column pa4 ba b--gray4 br2 z-1 bg-white"
```

Comment:
position absolute wtf @matildepark

Reply:
The dropdown list is supposed to be directly below the dropdown prompt without taking up space on the layer below — absolute drops it immediately below outside the box model. This seems like the way you'd do that at first glance here. Can you explain the wtf

Reply:
I think the `flex flex-column` classes would just be overwritten by the inline style here btw @ixv